### PR TITLE
Replace some TOC headings with raw html to fix sidebar.

### PR DESCRIPTION
**Pull request by alasdairwilson**
This partially solves #6989

The issue is caused by headings in the tutorial/how to page becoming top level toc entries, by turning these into raw html headings this behaviour is avoided.

The issue also has a comment regarding the coords TOC but this can't be tackled till after #6954

### CODE REVIEW

**Review by nabobalis — APPROVED** [review]

**Review by dstansby — CHANGES_REQUESTED** [review]
This seems a bit hacky, and not a sustainable solution if/when the pages get modified. What happens if the toctrees are moved above the first heading on the page instead? Maybe then they'll be treated as children of the page they're on, instead of children of a section within that page.

**Comment by Cadair** [comment]
> This seems a bit hacky

oh yes.

> not a sustainable solution if/when the pages get modified

Why specifically? Yes it's weird, but it's just inline content, we aren't breaking references or anything else with this hack.

> What happens if the toctrees are moved above the first heading on the page instead?

This works fine, but then you either have the whole contents list above the intro which is weird, or you don't have the contents list in the index page at all (i.e. make the toctree hidden).

**Comment by dstansby** [comment]
I was thinking more from the point of view of new contributiors (or old ones) coming along later and wondering why these aren't just RST headings.

Alternative fix, we could just drop the headings altogether? They don't seem super neccesary to have.

**Comment by Cadair** [comment]
My solution was going to be to make them bold.

**Comment by alasdairwilson** [comment]
One solution that came to mind would be to have a hidden toctree for the sidebar and then a manually generated listed that looks like the current section listing but this is hard to maintain because future changes to the toc or ordering of the sections aren't changed in both places automatically.

Despite this being hacky, I think this html solution is maintainable. I think it is clear even by a glance at what the sphinx directive is doing, inserting a heading into the generated html, and it would be pretty much impossible to break them regardless of where the toc on this page is moved. Someone might not be able to work out why regular rst section headings aren't being used but I don't think its that obtuse, certainly not the most egregious patch anyway, (and if they changed them to rst headings they would probably immediately clock the reasoning).

I don't mind the headings going (or going bold) but I do think that the page is better with the headings than without so the question is is the tradeoff of using this directive and perhaps being unclear worth it.

**Comment by dstansby** [comment]
I'd be 👍 just making them bold

**Comment by wtbarnes** [comment]
> I'd be 👍 just making them bold

I'm in this camp too I think. I think the headings are still obvious, especially since they are mostly at the top of the page anyway.

**Comment by wtbarnes** [comment]
Do we want to hold this up until #6954 is merged so it can fix those headings too?

**Review by dstansby — APPROVED** [review]

**Comment by nabobalis** [comment]
> Do we want to hold this up until #6954 is merged so it can fix those headings too?

I don't see why not, might as well do it all in one go.

**Review by Cadair — APPROVED** [review]

**Pull request merged**
